### PR TITLE
7858 UI changes to create a repairer's lien financing statement.

### PR DESCRIPTION
**Pull request by doug-lovett**
Signed-off-by: Doug Lovett <doug@diamante.ca>

*Issue #:* /bcgov/entity#7858

*Description of changes:*
Changes to create a Repairer's Lien financing statement.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/registration/RegistrationLengthTrust.vue [hunk 1]
```diff
@@ -155,7 +266,37 @@ export default defineComponent({
...
+        return ''
+      }),
+      surrenderDateMessage: computed((): string => {
+        if (lengthTrust.showInvalid && lengthTrust.surrenderDate === '') {
+          return 'This field is required'
+        }
+        return ''
+      }),
+      minSurrenderDate: computed((): string => {
+        var dateOffset = (24 * 60 * 60 * 1000) * 21
```
Comment (jordiwes) [inline ppr-ui/src/components/registration/RegistrationLengthTrust.vue]: perhaps a comment here on what this is :)

**Comment by jordiwes** [comment]
Looks good!

**Review by jordiwes — APPROVED** [review]

File: ppr-ui/src/components/registration/RegistrationLengthTrust.vue [hunk 2]
```diff
@@ -27,89 +32,186 @@
           {{lengthSummary}}
         </v-col>
       </v-row>
-      <v-row no-gutters class="ps-6 pb-6">
+      <v-row no-gutters class="ps-6 pb-6"  v-if="registrationType === 'SA'">
```
Comment (kialj876) [inline ppr-ui/src/components/registration/RegistrationLengthTrust.vue:35]: please use the enum in this comparison

File: ppr-ui/src/components/registration/RegistrationLengthTrust.vue [hunk 3]
```diff
@@ -27,89 +32,186 @@
...
         <v-col cols="3" class="generic-label">
           Trust Indenture
         </v-col>
         <v-col :class="$style['summary-text']">
           {{trustIndentureSummary}}
         </v-col>
       </v-row>
+      <v-row no-gutters class="ps-6 pb-3"  v-if="registrationType === 'RL'">
```
Comment (kialj876) [inline ppr-ui/src/components/registration/RegistrationLengthTrust.vue]: same thing for this one and below

File: ppr-ui/src/components/registration/RegistrationLengthTrust.vue [hunk 4]
```diff
@@ -133,6 +235,13 @@ export default defineComponent({
     const feeInfoInfinite = getFinancingFee(true)
     const router = context.root.$router
 
+    if (props.defaultRegistrationType === 'RL' && lengthTrust.lifeYears !== 1) {
```
Comment (kialj876) [inline ppr-ui/src/components/registration/RegistrationLengthTrust.vue]: enum again here please

File: ppr-ui/src/components/registration/RegistrationLengthTrust.vue [hunk 5]
```diff
@@ -155,7 +266,37 @@ export default defineComponent({
...
+        var dateOffset = (24 * 60 * 60 * 1000) * 21
+        var minDate = new Date()
+        minDate.setTime(minDate.getTime() - dateOffset)
+        return minDate.toISOString()
+      }),
+      maxSurrenderDate: computed((): string => {
+        return new Date().toISOString()
+      }),
       lengthSummary: computed((): string => {
+        if (localState.registrationType === 'RL') {
```
Comment (kialj876) [inline ppr-ui/src/components/registration/RegistrationLengthTrust.vue]: and here :)

File: ppr-ui/src/store/getters/state-getters.ts [hunk 6]
```diff
@@ -156,11 +156,16 @@ export const showStepErrors = (state: StateIF): boolean => {
  * Returns the array of steps.
  */
 export const getSteps = (state: any, getters: any): Array<any> => {
+  const regType:RegistrationTypeIF = getRegistrationType(state)
+  var lengthTrustText = 'Length and<br />Trust Indenture'
+  if (regType.registrationTypeAPI === 'RL') {
```
Comment (kialj876) [inline ppr-ui/src/store/getters/state-getters.ts]: enum please

File: ppr-ui/src/utils/registration-helper.ts [hunk 7]
```diff
@@ -40,6 +40,10 @@ export async function saveFinancingStatementDraft (stateModel:StateModelIF): Pro
   statement.lifeInfinite = trustLength.lifeInfinite
   statement.lifeYears = trustLength.lifeYears
   statement.trustIndenture = trustLength.trustIndenture
+  if (registrationType.registrationTypeAPI === 'RL') {
```
Comment (kialj876) [inline ppr-ui/src/utils/registration-helper.ts]: enum here and below please :)

File: ppr-ui/src/views/LengthTrust.vue [hunk 8]
```diff
@@ -16,16 +16,24 @@
             </v-row>
             <stepper class="mt-4" />
             <v-row no-gutters class="pt-10">
-              <v-col cols="auto" class="sub-header">
+              <v-col cols="auto" class="sub-header" v-if="registrationType !== 'RL'">
```
Comment (kialj876) [inline ppr-ui/src/views/LengthTrust.vue]: enum for this file too here and below (just adding these comments to make it easier to check them off)

File: ppr-ui/tests/unit/RegistrationHelper.spec.ts [hunk 9]
```diff
@@ -84,6 +97,19 @@ describe('Registration API Helper Tests', () => {
     expect(draft.financingStatement.documentId).toBeDefined()
   })
 
+  it('save new RL financing statement draft', async () => {
+    const testDraft = (JSON.parse(JSON.stringify(mockedDraftFinancingStatementAll)))
+    testDraft.financingStatement.type = 'RL'
```
Comment (kialj876) [inline ppr-ui/tests/unit/RegistrationHelper.spec.ts]: enum for here too

**Review by kialj876 — COMMENTED** [review]
Hey Doug, this looks very good to me. All my comments are just for swapping your registration type string comparisons/values with the enum values

File: ppr-ui/src/components/registration/RegistrationLengthTrust.vue (hunk 1 above)
Comment (doug-lovett) [reply ppr-ui/src/components/registration/RegistrationLengthTrust.vue]: Comment added

File: ppr-ui/src/components/registration/RegistrationLengthTrust.vue (hunk 2 above)
Comment (doug-lovett) [reply ppr-ui/src/components/registration/RegistrationLengthTrust.vue:35]: Updated in latest commit to use enum.

File: ppr-ui/src/components/registration/RegistrationLengthTrust.vue (hunk 3 above)
Comment (doug-lovett) [reply ppr-ui/src/components/registration/RegistrationLengthTrust.vue]: Updated in latest commit to use enum.

File: ppr-ui/src/components/registration/RegistrationLengthTrust.vue (hunk 4 above)
Comment (doug-lovett) [reply ppr-ui/src/components/registration/RegistrationLengthTrust.vue]: Updated in latest commit to use enum.

File: ppr-ui/src/components/registration/RegistrationLengthTrust.vue (hunk 5 above)
Comment (doug-lovett) [reply ppr-ui/src/components/registration/RegistrationLengthTrust.vue]: Updated in latest commit to use enum.

File: ppr-ui/src/store/getters/state-getters.ts (hunk 6 above)
Comment (doug-lovett) [reply ppr-ui/src/store/getters/state-getters.ts]: Updated in latest commit to use enum.

File: ppr-ui/src/utils/registration-helper.ts (hunk 7 above)
Comment (doug-lovett) [reply ppr-ui/src/utils/registration-helper.ts]: Updated in latest commit to use enum.

File: ppr-ui/src/views/LengthTrust.vue (hunk 8 above)
Comment (doug-lovett) [reply ppr-ui/src/views/LengthTrust.vue]: Updated in latest commit to use enum.

File: ppr-ui/tests/unit/RegistrationHelper.spec.ts (hunk 9 above)
Comment (doug-lovett) [reply ppr-ui/tests/unit/RegistrationHelper.spec.ts]: Updated in latest commit to use enum.

**Review by jordiwes — APPROVED** [review]

**Review by kialj876 — APPROVED** [review]

**Pull request merged**